inst/templates/ad_adlb.R

```diff
@@ -0,0 +1,331 @@
+    by_vars = vars(LBTESTCD)
+  ) %>%
+
+  # Calculate PARCAT1 AVAL AVALC ANRLO ANRHI
+  mutate(
+    PARCAT1 = LBCAT,
+    AVAL  = LBSTRESN,
+    AVALC = LBSTRESC,
+    ANRLO = LBSTNRLO,
+    ANRHI = LBSTNRHI
+  )
+
+# Get Visit Info
+adlb <- adlb %>%
+
+  # Derive Timing
+  mutate(
+    AVISIT = case_when(
+      str_detect(VISIT, "SCREEN") ~ "Baseline",
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    ),
+    AVISITN = case_when(
+      AVISIT == "Baseline" ~ 0,
+      !is.na(VISITNUM) ~ VISITNUM
+    )
+  )
+
+# Get WBC
+adlb <- adlb %>%
+  derive_param_wbc_abs(by_vars = vars(USUBJID, VISIT),
+                       set_values_to = vars(PARAMCD = "LYMPH",
+                                            PARAM = "Leukocytes (GI/L)",
+                                            DTYPE = "CALCULATION"),
+                       get_unit_expr = extract_unit(PARAM),
+                       wbc_code = "WBC",
+                       diff_code = "LYMLE",
+                       diff_type = "fraction",
+                       wbc_unit = "GI/L")
+
+adlb <- adlb %>%
+
+  # Calculate ONTRTFL
+  derive_var_ontrtfl(
+    start_date = ADT,
+    ref_start_date = TRTSDT,
+    ref_end_date = TRTEDT,
+    filter_pre_timepoint = AVISIT == "Baseline"
+  )
+
+# Calculate ANRIND : requires the reference ranges ANRLO, ANRHI
+adlb <- adlb %>%
+  derive_var_anrind()
+
+# Derive baseline flags
+adlb <- adlb %>%
+  # Calculate BASETYPE
+  mutate(
+    BASETYPE = "LAST"
+  ) %>%
+
+  # Calculate ABLFL
+  derive_var_extreme_flag(
+    by_vars = vars(STUDYID, USUBJID, BASETYPE, PARAMCD),
+    order = vars(ADT, VISITNUM, LBSEQ),
+    new_var = ABLFL,
+    mode = "last",
+    filter = (!is.na(AVAL) & ADT <= TRTSDT & !is.na(BASETYPE))
+  )
+
+# Derive baseline information
+adlb <- adlb %>%
+
+  # Calculate BASE
+  derive_var_base(
+    by_vars = vars(STUDYID, USUBJID, PARAMCD, BASETYPE),
+    source_var = AVAL,
+    new_var = BASE
+  ) %>%
+
+  # Calculate BASEC
+  derive_var_base(
+    by_vars = vars(STUDYID, USUBJID, PARAMCD, BASETYPE),
+    source_var = AVALC,
+    new_var = BASEC
+  ) %>%
+
+  # Calculate BNRIND
+  derive_var_base(
+    by_vars = vars(STUDYID, USUBJID, PARAMCD, BASETYPE),
+    source_var = ANRIND,
+    new_var = BNRIND
+  ) %>%
+
+  # Calculate CHG
+  derive_var_chg() %>%
+
+  # Calculate PCHG
+  derive_var_pchg()
+
+
+# R2BASE, R2ANRLO R2ANRHI
+adlb <- adlb %>%
+  derive_var_analysis_ratio(
+    numer_var = AVAL,
+    denom_var = BASE
+  ) %>%
+  derive_var_analysis_ratio(
+    numer_var = AVAL,
+    denom_var = ANRLO
+  ) %>%
+  derive_var_analysis_ratio(
+    numer_var = AVAL,
+    denom_var = ANRHI
+  )
+
+# SHIFT derivation
+adlb <- adlb %>%
+  derive_var_shift(
+    new_var = SHIFT1,
+    from_var = BNRIND,
+    to_var = ANRIND
+    )
+
+# ANL01FL: Flag last result within an AVISIT for post-baseline records
+# LVOTFL: Flag last valid on-treatment record
+adlb <- adlb %>%
+  derive_var_extreme_flag(
```

Comment:
Same as above (for ABLFL calculation) use `restrict_derivation` to filter.

Reply:
Updated